### PR TITLE
Add `edit` and `delete` methods for Message and handle `MessageUpdate` and `MessageDelete` events

### PR DESCRIPTION
**Pull request by rockboy987**

### CODE REVIEW

File: revolt/types/gateway.py [hunk 1]
```diff
@@ -32,3 +34,16 @@ class ReadyEventPayload(BasePayload):
 
 class MessageEventPayload(BasePayload, Message):
     pass
+
+class MessageUpdateData(TypedDict):
+    content: str
+    edited: dict[str, Any]
```
Comment (Zomatree) [inline revolt/types/gateway.py:40]: Is this able to be narrowed down even more so `Any` isn't needed? A typeddict here would be prefered

File: revolt/types/gateway.py [hunk 2]
```diff
@@ -16,6 +16,8 @@
     "AuthenticatePayload",
     "ReadyEventPayload",
     "MessageEventPayload",
+    "MessageUpdateEventPayload",
+    "MessageDeleteEventPayload",
 )
```
Comment (Zomatree) [inline revolt/types/gateway.py:21]:
```suggestion
    "MessageUpdateData",
 )
```

**Review by Zomatree — CHANGES_REQUESTED** [review]

File: revolt/types/gateway.py (hunk 1 above)
Comment (rockboy987) [reply revolt/types/gateway.py]: yup, just fixed this sorry

**Comment by Zomatree** [comment]
LGTM

**Pull request merged**